vllm/spec_decode/batch_expansion.py

```diff
@@ -292,6 +292,20 @@ def _create_single_target_seq_group_metadata(
         for data in new_seq_data_dict.values():
             data.update_num_computed_tokens(data.get_len() - 1)
 
+
+        if seq_group_metadata.state is not None and seq_group_metadata.state.generator is not None:
+            generator = torch.Generator(
+                device=seq_group_metadata.state.generator.device
+            )
+            generator.set_state(
+                seq_group_metadata.state.generator.get_state()
+            )
+            state = SequenceGroupState(
+                generator=generator,
+            )
```

Comment:
Could the same Generator just be used here?

Reply:
My first attempt just shallow copied the generator but it led to incorrect results, so I think we need a "deep" copy here.

Reply:
It's still not obvious to me why this doesn't work, we could look into it as a follow-on because it's not ideal to create a new generator every iteration.